client/network/src/config.rs

```diff
@@ -643,6 +723,97 @@ impl NonReservedPeerMode {
 	}
 }
 
+///Doc
+#[derive(Clone, Debug)]
+pub enum PreShareKeyConfig {
+	/// A pre-shared key configuration.
+	PRESHAREDKEY(PreSharedKeySecret),
+}
+
+/// The options for obtaining a Ed25519 secret key.
+pub type PSKey = PreSharedKeySecret;
+
+///Doc
+#[derive(Clone)]
+pub enum PreSharedKeySecret {
+	/// Read the pre-shared key from a file.
+	File(PathBuf),
+	///Doc
+	Hard,
+}
+
+impl Default for PreShareKeyConfig {
+	fn default() -> PreShareKeyConfig {
+		Self::PRESHAREDKEY(PreSharedKeySecret::Hard)
+	}
+}
+
+impl fmt::Debug for PreSharedKeySecret {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		match self {
+			Self::File(path) => f.debug_tuple("PSKSecret::File").field(path).finish(),
+			Self::Hard => f.debug_tuple("PSKSecret::Hard").finish(),
+		}
+	}
+}
+
+impl PreShareKeyConfig {
+	/// Evaluate a `PreSharedKeyConfig` to obtain an identity `pre shared key`:
+	///
+	///  * If the secret is configured as a file, it is read from that file, if it exists. Otherwise
+	///    return error.
+	pub fn into_pre_share_key(self) -> io::Result<PreSharedKey> {
+		use PreShareKeyConfig::*;
+		match self {
+			PRESHAREDKEY(PreSharedKeySecret::Hard) => {
+				Ok(
+				PreSharedKey::new(
+					[24, 97, 125, 255, 78, 254, 242, 4, 80, 221, 94, 175, 192, 96, 253,
+					133, 250, 172, 202, 19, 217, 90, 206, 59, 218, 11, 227, 46, 70, 148, 252, 215]
+				))
+			},
+			PRESHAREDKEY(PreSharedKeySecret::File(f)) => get_pre_shared_key(
+					f, 
+					|b| match String::from_utf8(b.to_vec()).ok().and_then(|s| {
```

Comment:
I think we can simplify this function a lot. Don't use the function with generics, just read and parse the key here